### PR TITLE
[#789] Use Inter font in Zashi

### LINKED ISSUE #789: Use Inter font in Zashi
All labels should use custom font:

https://github.com/rsms/inter

### PR DESCRIPTION
**Pull request by LukasKorba**
Closes #789 

- Inter fonts added and registered for use
- all Texts in the Zashi use Inter as a default font
- some texts have custom sizes
- cleanup of fonts in the project

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [x] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [x] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: modules/Sources/Generated/Fonts+Generated.swift [hunk 1]
```diff
@@ -19,43 +19,32 @@ public typealias SystemFont = FontConvertible.SystemFont
...
+    public static let medium = FontConvertible(name: "Inter-Medium", family: "Inter", path: "Inter-Medium.otf")
+    public static let mediumItalic = FontConvertible(name: "Inter-MediumItalic", family: "Inter", path: "Inter-MediumItalic.otf")
+    public static let regular = FontConvertible(name: "Inter-Regular", family: "Inter", path: "Inter-Regular.otf")
+    public static let semiBold = FontConvertible(name: "Inter-SemiBold", family: "Inter", path: "Inter-SemiBold.otf")
+    public static let semiBoldItalic = FontConvertible(name: "Inter-SemiBoldItalic", family: "Inter", path: "Inter-SemiBoldItalic.otf")
+    public static let thin = FontConvertible(name: "Inter-Thin", family: "Inter", path: "Inter-Thin.otf")
+    public static let thinItalic = FontConvertible(name: "Inter-ThinItalic", family: "Inter", path: "Inter-ThinItalic.otf")
+    public static let all: [FontConvertible] = [black, blackItalic, bold, boldItalic, extraBold, extraBoldItalic, extraLight, extraLightItalic, italic, light, lightItalic, medium, mediumItalic, regular, semiBold, semiBoldItalic, thin, thinItalic]
```
Comment (HonzaR) [inline modules/Sources/Generated/Fonts+Generated.swift:41]: Do we use all of these fonts in the project? E.g., the italic variants might be unnecessary. On the Android side, we use downloadable fonts, so once we approach to use of a new variant, it's downloaded in runtime. So, we don't pack in any unused resources.

File: modules/Sources/Features/BalanceBreakdown/BalanceBreakdownView.swift [hunk 2]
```diff
@@ -77,8 +77,10 @@ extension BalanceBreakdownView {
                     tokenName
                 )
             )
-            .font(.system(size: 32))
-            .fontWeight(.bold)
+            .font(
+                .custom(FontFamily.Inter.regular.name, size: 32)
```
Comment (HonzaR) [inline modules/Sources/Features/BalanceBreakdown/BalanceBreakdownView.swift:81]: This is out of the scope of this task, but it's still worth asking. I see a pattern of adding this extra font parameter to many texts on different screens. Could we extra commonly set parameters as default parameters to a new UI component which we then use on several screens?

Comment (HonzaR) [reply modules/Sources/Features/BalanceBreakdown/BalanceBreakdownView.swift:81]: And second related question, why some of the UI text components here do not use the font parameter like the one on line 72?

**Review by HonzaR — COMMENTED** [review]
It looks good to me. I just left a few comments inline.

File: modules/Sources/Generated/Fonts+Generated.swift (hunk 1 above)
Comment (LukasKorba) [reply modules/Sources/Generated/Fonts+Generated.swift:41]: I simply used the whole font family and did not do any selection. I think in theory we use 2 or 3 out of 19. We don't have any dynamic downloading so it's much easier to use the whole font family than go through the whole process of adding one extra when needed.

File: modules/Sources/Features/BalanceBreakdown/BalanceBreakdownView.swift (hunk 2 above)
Comment (LukasKorba) [reply modules/Sources/Features/BalanceBreakdown/BalanceBreakdownView.swift:81]: > This is out of the scope of this task, but it's still worth asking. I see a pattern of adding this extra font parameter to many texts on different screens. Could we extra commonly set parameters as default parameters to a new UI component which we then use on several screens?

There's not much I can do except making a view modifier that does almost the same, except the font. For example:
```
.interFont(32)
```

Maybe nice to have, I can create a followup ticket.

Comment (LukasKorba) [reply modules/Sources/Features/BalanceBreakdown/BalanceBreakdownView.swift:81]: > And second related question, why some of the UI text components here do not use the font parameter like the one on line 72?

The font is propagated from the parent. I set default Inter font to the root of the app so all Text instances use Inter in a default setup. Whenever needed, this must be overloaded by local .font instance.

**Review by HonzaR — APPROVED** [review]

**Pull request merged**